icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/DateTimeGeneratorTest.java

```diff
@@ -1793,4 +1793,59 @@ public void test_jConsistencyOddLocales() { // ICU-20590
             }
         }
     }
+    
+    @Test
+    public void testBestPattern() {
+        // generic test for DateTimePatternGenerator::getBestPattern() that can be used to test multiple
+        // bugs in the resource data
+        String[] testCases = {
+            // ICU-21650: if the user asks for E, the minimum field length is 3, but if he asks for c or e, it's 1
```

Comment:
same comment as C++, please add more comment to explain. Thanks

Reply:
See above.  I added a comment to the test that points to UTS 35.